File: TypeScript/codewars/Typescript/Encrypt-this.ts
export const encryptThis = (str: string): string => {
	if (!str.trim()) return "";

	return str
		.split(" ")
		.map((word) => {
			if (word.length === 0) return "";

			const firstLetterAscii = word[0].charCodeAt(0);
			if (word.length === 1) return firstLetterAscii.toString();
			if (word.length === 2) return firstLetterAscii + word[1];

			const middlePart = word.slice(2, word.length - 1);
			return firstLetterAscii + word[word.length - 1] + middlePart + word[1];
		})
		.join(" ");
};

File: Csharp/CodeWars/src/Completed/EncryptThis.cs
namespace CodeWars.Completed;

public class EncryptThisClass
{
    public static string EncryptThis(string input)
    {
        if (string.IsNullOrWhiteSpace(input)) return string.Empty;

        return string.Join(" ", input.Split(' ')
            .Select(word =>
            {
                if (string.IsNullOrEmpty(word)) return "";

                int ascii = word[0];

                if (word.Length == 1)
                    return ascii.ToString();

                if (word.Length == 2)
                    return ascii + word[1].ToString();

                string second = word[1].ToString();

                var last = word[^1].ToString();
                string middle = word.Length > 3 ? word[2..^1] : "";

                return ascii + last + middle + second;
            }));
    }
}
